clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -1135,16 +1137,42 @@ scheduler_tmpl_t<RecordType, ReaderType>::read_traced_schedule()
     res = remove_zero_instruction_segments(input_sched, all_sched);
     if (res != sched_type_t::STATUS_SUCCESS)
         return res;
-    for (int output_idx = 0; output_idx < static_cast<output_ordinal_t>(outputs_.size());
-         ++output_idx) {
+    // Sort by cpuid to get a more natural ordering.
+    // Probably raw2trace should do this in the first place, but we have many
+    // schedule files already out there so we still need a sort here.
+    std::sort(disk2index.begin(), disk2index.end(),
+              [disk2cpuid](const output_ordinal_t &l, const output_ordinal_t &r) {
+                  return disk2cpuid[l] < disk2cpuid[r];
+              });
+    // disk2index now holds the sorted index, but we need another array to
+    // store that in the disk order. E.g., if the original file held 6,2,3,7,
+    // disk2index would then hold 1,2,0,3, but we want 2,0,1,3.
```

Comment:
This example should be improved. Each cpu can of course have multiple schedule entries. So disk2index may have repeated elements.

Reply:
Understood that it cannot. The example is fine.